feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/repository/WalletRepositoryImpl.kt

```diff
@@ -84,41 +84,56 @@ class WalletRepositoryImpl(
         return syncAssetsRates(account)
     }
 
-    override fun transactionsFirstPageFlow(currentAccount: WalletAccount, pageSize: Int, accounts: List<WalletAccount>): Flow<List<Transaction>> {
-        return observeTransactions(currentAccount, accounts)
+    override fun operationsFirstPageFlow(currentAccount: WalletAccount, accounts: List<WalletAccount>): Flow<List<Operation>> {
+        val accountsByAddress = accounts.associateBy { it.address }
+
+        return operationDao.observe(currentAccount.address).mapList {
+            val accountName = defineAccountNameForTransaction(accountsByAddress, displayAddress = it.receiver ?: it.sender ?: it.address)
```

Comment:
displayAddress = it.receiver ?: it.sender ?: it.address - is it the right place to put this logic? Probably it should be determined on mapping from db to domain (domain mapper)

Reply:
Moved logic inside defineAccountNameForTransaction method